### PR TITLE
Add a flag to disable default launch plan generation

### PR DESCRIPTION
**Pull request by EngHabu**
# TL;DR
Adds a flag to control whether a default launchplan should be generated for the workflow at registration time. The default behavior is to do so.

E.g.
```
@workflow_class(disable_default_launch_plan=True)
def MyWorkflow():
   ...

my_lp = MyWorkflow.create_launch_plan(...)

#my_lp will be the only launchplan created for this workflow
```

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [x] Code documentation added
 - [x] Any pending items have an associated Issue

## Tracking Issue
https://github.com/lyft/flyte/issues/462

### CODE REVIEW

**Comment by EngHabu** [comment]
PTAL @wild-endeavor

File: flytekit/__init__.py [hunk 1]
```diff
@@ -2,4 +2,4 @@
 
 import flytekit.plugins  # noqa: F401
 
-__version__ = "0.12.4"
+__version__ = "0.12.5"
```
Comment (RubenBarragan) [inline flytekit/__init__.py:5]: nit question, do we have to change the version every change?

**Review by RubenBarragan — APPROVED** [review]

File: flytekit/__init__.py (hunk 1 above)
Comment (EngHabu) [reply flytekit/__init__.py:5]: yes

**Comment by honnix** [comment]
Looks like this feature is gone. Is there a replacement? Thanks.

**Pull request merged**